torch/onnx/_internal/fx/fx_onnx_interpreter.py

```diff
@@ -294,100 +234,194 @@ def _wrap_fx_args_as_onnxscript_args(
+        self.onnxfunction_dispatcher = onnxfunction_dispatcher
+
+        # At `run`, a TorchScript graph is augmented to
+        # represent the input FX graph with ONNX symbols (e.g., onnx::add).
+        # To connect the values to nodes in the TorchScript graph, we maintain
+        # fx_name_to_onnxscript_value. Basically, we want to translate
+        #   fx_tensor_x (type: torch.fx.Node) -> fx_node_1 -> fx_tensor_y (type: torch.fx.Node)
+        # to
+        #   fx_name_to_onnxscript_value[fx_tensor_x.name] -> onnx_node_1 -> fx_name_to_onnxscript_value[fx_tensor_y.name]
+
+        self.fx_name_to_onnxscript_value: Dict[
+            str,
+            Union[
+                onnxscript_graph_builder.TorchScriptTensor,
+                Tuple[onnxscript_graph_builder.TorchScriptTensor, ...],
+            ],
+        ] = {}
+        self.tracer = None
+        self.op_level_debug = op_level_debug
+        self.diagnostic_context = diagnostic_context
+
+        # TODO: Refactor where/how onnxscript_graph is store/passed in
+        self.tracer = onnxscript_graph_builder.TorchScriptTracingEvaluator(
+            self.onnxscript_graph
         )
-        location = _location_from_fx_stack_trace(node_stack_trace)
-        if location is not None:
-            diagnostic.with_location(location)
 
-    if node.op == "placeholder":
+    @_beartype.beartype
+    def run(self):
+        """Analyze the specified FX node and trigger the ONNX translation for it.
+
+        Args:
+            node: The TorchFX node to dispatch the function for.
+            onnx_args: The arguments of the ONNX function.
+            onnx_kwargs: The keyword arguments of the ONNX function.
+            diagnostic_context: The diagnostic context to use for reporting errors.
+
+        Returns:
+            Either an `onnxscript.OnnxFunction` or `onnxscript.TracedOnnxFunction` instance based on the dispatch algorithm.
+
+        Raises:
+            RuntimeError: If there are no overloaded functions available for the given FX node.
+        """
+
+        # TODO: Fix FakeTensorMode limitation asap
+        # We want to pass list of ints and floats to TorchScript graph correctly
+        # in _export_fx_to_ts, so we must disable FakeTensorMode. Otherwise, graph may
+        # receive FakeTensor and results runtime error. In addition, TorchScript-based
+        # ONNX exporter used in _ts_graph_to_onnx_model_in_protobuf is not compatible
+        # with FakeTensorMode.
+        with torch.utils._mode_utils.no_dispatch():
+            # node_fixed_shape is only used on op_level_debug purpose.
+            for node in self.graph_module.graph.nodes:
+                # Dispatch to ONNX op through OpSchema. The input argument dtypes are compared to
+                # function signature in OpSchema, and find the best matched overload.
+                # TODO(titaiwang): diagnostic rules.
+
+                if node.op == "placeholder":
+                    self.placeholder(node)
```

Comment:
As mentioned, this class should only contain `staticmethod`s for making dataflow as explicit as possible. `self.placeholder` and similar calls will make the code as stateful as writing a `fx.Interpreter`.

Reply:
as above